XMonad/Actions/WindowBringer.hs

```diff
@@ -45,83 +46,106 @@ import XMonad.Util.NamedWindows (getName)
 -- For detailed instructions on editing your key bindings, see
 -- "XMonad.Doc.Extending#Editing_key_bindings".
 
--- | Default menu command
-defaultCmd :: String
-defaultCmd = "dmenu"
+data WindowBringerConfig = WindowBringerConfig { menuCommand :: String
+                                               , menuArgs :: [String]
+                                               , windowTitler :: X.WindowSpace -> Window -> X String
+                                               }
 
--- | Make dmenu case insensitive
-defaultArgs :: [String]
-defaultArgs = ["-i"]
+instance Default WindowBringerConfig where
+    def = WindowBringerConfig{ menuCommand="dmenu"
```

Comment:
It's customary style to put spaces on either side of `=`.  Personally, I think it also looks nice to vertically align the `=` signs (though it's not necessary).